Game.py

```diff
@@ -1,46 +1,67 @@
-# -*- coding: utf-8 -*-
-
-###########################################
-# redundant file atm
-###########################################
-
-
-
-"""
-simulate the pirsoners dilemma
-"""
+import abc, random, numpy as np
 from enum import Enum
-import numpy as np
 
 class Strategy(Enum):
     cooperate = 0
     defect = 1
 
 class Player():
-    def __init__(self, strategy, x, y, p):
+    def __init__(self, strategy):
         self.strategy = strategy
-        self.i = x  # vertical position
-        self.j = y  # horizontal position
-        self.money = 0
+        self.cell = None
+        self.payoff = 0
+
+class PrisonersDilemmaPlayer(Player):
+    def __init__(self, strategy):
+        super().__init__(strategy)
+
+
+class AsymmetricPrisonersDilemmaPlayer(Player):
+    def __init__(self, strategy):
+        super().__init__(strategy)
         self.physio = np.random.choice(np.arange(0,2), p=[p, 1-p])
         self.safety = self.physio*np.random.choice(np.arange(0,2), p=[p, 1-p])
         self.love = self.safety*np.random.choice(np.arange(0,2), p=[p, 1-p])
         self.esteem = self.love*np.random.choice(np.arange(0,2), p=[p, 1-p])
         self.fulfill = self.esteem*np.random.choice(np.arange(0,2), p=[p, 1-p])
```

Comment:
where does the p value come from?

Reply:
Good catch! I forgot to carry that over as a constructor argument. I'll add it in.